src/main/java/org/mcsr/speedrunapi/mixin_plugin/SpeedrunMixinConfigPlugin.java

```diff
@@ -31,9 +31,14 @@ public String getRefMapperConfig() {
 
     @Override
     public boolean shouldApplyMixin(String targetClassName, String mixinClassName) {
+        // ensure we do not force apply any mixin when the target class isn't present, otherwise causing a crash
+        if (SpeedrunMixinConfigPlugin.class.getClassLoader().getResource(targetClassName.replace('.', '/') + ".class") == null) {
+            System.out.println("target class " + targetClassName + " for mixin " + mixinClassName + " not found");
+            return false;
+        }
         String compatPackage = this.mixinPackage + ".compat.";
         if (mixinClassName.startsWith(compatPackage)) {
-            String modid = mixinClassName.replaceFirst(compatPackage, "").split("\\.", 2)[0];
+            String modid = mixinClassName.replaceFirst(compatPackage.replace(".", "\\."), "").split("\\.", 2)[0];
```

Comment:
I dont really see a reason for this change, if you do want to simplify it it could be rewritten to something like this:
```suggestion
            String modid = mixinClassName.substring(compatPackage.length()).split("\\.", 2)[0];
```

Reply:
the dots in mixinPackage and the ".compat." string are currently handled as wildcards, this doesn't really fix anything it's just to correct intent

Reply:
substring is probably better anyway though, do you want me to commit that?

Reply:
Ah that makes sense, yea i think just commiting the substring is probably the best solution